### PR TITLE
python 3.7.1

### PR DESCRIPTION
**Pull request by lembacon**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/python.rb [hunk 1]
```diff
@@ -22,15 +23,6 @@ class Python < Formula
     satisfy { MacOS::CLT.installed? }
   end
 
-  head do
```
Comment (Noctem) [inline Formula/python.rb]: Why did you remove this?

**Comment by lembacon** [comment]
We're having some troubles with the `sphinx-build` which is currently building against the system provided Python 2 framework.

The Sphinx configuration for documentations in Python 3 now relies on one of its own extensions named [`escape4chm`](https://github.com/python/cpython/blob/859c068e52a31e13e2b9bb6a3f861fa8c290cb0e/Doc/conf.py#L16-L17), the `escape4chm` depends on the [`html.entities`](https://github.com/python/cpython/blob/859c068e52a31e13e2b9bb6a3f861fa8c290cb0e/Doc/tools/extensions/escape4chm.py#L9) module which is not available in Python 2. We cannot simply make `sphinx-doc` build against our Python 3 formula, it will otherwise become a chicken-and-egg problem.

Vendoring `sphinx-doc` inside `python` seems the appropriate solution for now, unless we decide not to build documentations for `python`. 🙃

**Comment by cclauss** [comment]
Could we:
```python
for six.moves import html_entities
# or if no six...
try:
    import html.entities as html_entities
except ImportError:
    import htmlentitydefs as html_entities
```

**Comment by lembacon** [comment]
Adding another 20 resources for building documentations seems ridiculous to me at this point.

**Comment by commitay** [comment]
> Adding another 20 resources for building documentations seems ridiculous to me at this point.

Yeah

> I'd suggest dropping building those HTML documentations for Python 3.

👍

**Comment by fxcoudert** [comment]
Agree with not shipping the docs anymore. We could possibly have a `python-docs` package, that depends on python, if we really want to ship them at some point.

**Comment by fxcoudert** [comment]
- `arcade-learning-environment` linkage is longstanding issue
- `mps-youtube` is unrelated, and now fixed
- `pipenv` is worrying:

```
==> /usr/local/Cellar/pipenv/2018.10.13_1/bin/pipenv
Traceback (most recent call last):
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==2018.10.13', 'console_scripts', 'pipenv')()
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pkg_resources/__init__.py", line 484, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2714, in load_entry_point
    return ep.load()
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2332, in load
    return self.resolve()
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2338, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/__init__.py", line 23, in <module>
    from .cli import cli
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/cli/__init__.py", line 3, in <module>
    from .command import cli
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/cli/command.py", line 7, in <module>
    import crayons
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/patched/crayons.py", line 49, in <module>
    is_powershell = "powershell" in shellingham.detect_shell()[0]
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/vendor/shellingham/__init__.py", line 22, in detect_shell
    shell = get_shell(pid, max_depth=max_depth)
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/vendor/shellingham/posix/__init__.py", line 54, in get_shell
    mapping = _get_process_mapping()
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/vendor/shellingham/posix/__init__.py", line 15, in _get_process_mapping
    mapping = impl.get_process_mapping()
  File "/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv/vendor/shellingham/posix/ps.py", line 18, in get_process_mapping
    'ps', '-ww', '-o', 'pid=', '-o', 'ppid=', '-o', 'args=',
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 389, in check_output
    **kwargs).stdout
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 481, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['ps', '-ww', '-o', 'pid=', '-o', 'ppid=', '-o', 'args=']' returned non-zero exit status 1.
```

- `fb-client` might be a missing revision bump?

```
==> /usr/local/Cellar/fb-client/2.0.3_4/bin/fb -h
Traceback (most recent call last):
  File "/usr/local/Cellar/fb-client/2.0.3_4/libexec/bin/fb", line 15, in <module>
    import pycurl
ImportError: dlopen(/usr/local/Cellar/fb-client/2.0.3_4/libexec/vendor/lib/python3.7/site-packages/pycurl.cpython-37m-darwin.so, 2): Library not loaded: @@HOMEBREW_PREFIX@@/opt/openssl/lib/libcrypto.1.0.0.dylib
  Referenced from: /usr/local/Cellar/fb-client/2.0.3_4/libexec/vendor/lib/python3.7/site-packages/pycurl.cpython-37m-darwin.so
  Reason: image not found
Error: fb-client: failed
```

- `shared-mime-info` is having trouble on all three macOS version (it's a dependency of `evince`):

```
==> Pouring shared-mime-info-1.10.mojave.bottle.tar.gz
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall shared-mime-info`
```

**Comment by lembacon** [comment]
Both `fb-client` and `shared-mime-info` are long standing issues.

Testing failure of `pipenv` only occurs in Mojave, which is possibly due to some recent changes to our Mojave boxes and `ps` therefore cannot retrieve the process list. Actually, upstream's upstream has already fixed this. https://github.com/sarugaku/shellingham/commit/4f219daf8a11fc638555f1874b4286120e99699a

**Comment by tschoonj** [comment]
I am fixing the `evince`/`shared-mime-info` trouble in #33307

**Comment by pavelrad** [comment]
Is there any update? :)

**Comment by cainmatt** [comment]
It looks like the build-bot is complaining about the sha256 for the resource "docs-html" in Formula/python.rb.

The build-bot thinks the correct value is:
"62dd86cecfbb53c04b5d6f9ae213808b1e653c16c6f355f65423eaf751ec0f02"

**Comment by commitay** [comment]
As the docs tarball is unstable let's just drop the docs again so we can get this shipped.

**Comment by fxcoudert** [comment]
hi @lembacon any update on this?

**Comment by lembacon** [comment]
@fxcoudert Missed https://github.com/Homebrew/homebrew-core/pull/33224#issuecomment-434846916 from 6 days ago 😂, just pushed a forced update.

**Comment by fxcoudert** [comment]
@lembacon go ahead!

**Review by fxcoudert — APPROVED** [review]
